model/src/main/kotlin/Provenance.kt

```diff
@@ -43,7 +43,16 @@ data class Provenance(
          * The VCS repository that was downloaded, or null.
          */
         @JsonInclude(JsonInclude.Include.NON_NULL)
-        val vcsInfo: VcsInfo? = null
+        val vcsInfo: VcsInfo? = null,
+
+        /**
+         * The original [VcsInfo] that was used to download the source code. It can be different to [vcsInfo] if any
+         * automatic detection took place. For example if the original [VcsInfo] does not contain any revision and the
+         * revision was automatically detected by searching for a tag that matches the version of the package there
+         * would be no way to match the package to the [Provenance] without downloading the source code and searching
+         * for the tag again.
+         */
+        val originalVcsInfo: VcsInfo? = null
```

Comment:
Should this also have `@JsonInclude(JsonInclude.Include.NON_NULL)`?

Reply:
Yes, thanks! Done.